packages/core/src/sfpcommands/source/DeploySourceToOrgImpl.ts

```diff
@@ -215,72 +115,94 @@ export default class DeploySourceToOrgImpl implements DeploymentExecutor {
         SFPLogger.log(table.toString(), LoggerLevel.WARN, this.packageLogger);
     }
 
-    private displayTestFailures(testFailures) {
+    private displayTestFailures(testFailures: Failures | Failures[]) {
         let table = new Table({
             head: ['Test Name', 'Method Name', 'Message'],
         });
 
-        testFailures.forEach((elem) => {
-            table.push([elem.name, elem.methodName, elem.message]);
-        });
-
+        if (Array.isArray(testFailures)) {
+            testFailures.forEach((elem) => {
+                table.push([elem.name, elem.methodName, elem.message]);
+            });
+        } else {
+            table.push([testFailures.name, testFailures.methodName, testFailures.message]);
+        }
         SFPLogger.log('Unable to deploy due to test failures:', LoggerLevel.WARN, this.packageLogger);
         SFPLogger.log(table.toString(), LoggerLevel.WARN, this.packageLogger);
     }
 
-    private buildExecCommand(): string {
-        let apexclasses;
-
-        let command = `sfdx force:mdapi:deploy -u ${this.target_org}`;
-
-        if (this.deployment_options['checkonly']) command += ` -c`;
-
-        //directory
-        command += ` -d ${this.mdapiDir}`;
-
-        //add json
-        command += ` --json`;
+    private async buildDeploymentOptions(sourceDir: string, username: string): Promise<MetadataApiDeployOptions> {
+        username = await convertAliasToUsername(username);
+        let metdataDeployOptions: MetadataApiDeployOptions = {
+            usernameOrConnection: username,
+            apiOptions: {},
```

Comment:
Additional apiOptions: 
- rollbackOnError
- ignoreWarnings
- rest

Reply:
Handled elsewhere depending on the conditions